pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -145,6 +147,13 @@ func (builder *ListenerBuilder) aggregateVirtualInboundListener() *ListenerBuild
 	// TODO: Trim the inboundListeners properly. Those that have been added to filter chains should
 	// be removed while those that haven't been added need to remain in the inboundListeners list.
 	filterChains, needTLS := reduceInboundListenerToFilterChains(builder.inboundListeners)
+	sort.SliceStable(filterChains, func(i, j int) bool {
```

Comment:
@kyessenov What scenarios do we need to sort? 
I am a little confused by what are sorted by serialization framework and what are not.

Reply:
The order always matters in terms of Envoy determining if it should reload the config, but Envoy doesn't necessarily rely on the order (like here). So any case where the order doesn't matter we need to make sure the order is consistent

Reply:
At a minimum, we need stable sorting for Virtual Inbound listener which now has multiple filter chains for inbound ports.

I'm planning to update this PR to add names to all filter chains and always sort. Why should we not sort everywhere?

Reply:
Any list or repeated field needs to be sorted. Deterministic protobuf marshal only guarantees that maps will be pre-sorted before writing them on the wire.

Reply:
Aha, only map is default sorted... Thanks!

Reply:
@nrjpoddar @howardjohn 
Since envoy run hash for each listener: every repeated field in listener fields and sub fields need to be sorted.